### ISSUE TITLE
All products are assumed to be Product_Observational by PLAID

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS/PLAID#25]
## 🐛 Describe the bug

Whenever I try to create a product not Observational (e.g. Document, Context, etc.), the software tries to drop this into `Product_Observational` and it does not work.

Minimal support should include:
See https://pds.nasa.gov/datastandards/documents/im/v1/index_1G00.html#class_pds_observation_area for:
* Product_Observational

See https://pds.nasa.gov/datastandards/documents/im/v1/index_1G00.html#class_pds_context_area for:
* Product_Context
* Product_Document
* Product_Collection
* Product_Bundle

**Comment by jordanpadams** [comment]
bug may be here: https://github.com/NASA-PDS/PLAID/blob/2511e5cd6e40499e5ce7edca0699bb74cc92eab8/php/interact_db.php#L570

**Comment by voxparcxls** [comment]
@jordanpadams  

The two criteria for Issue#25 to include 2 product types (Collection, Bundle) and fix XML mutator to ascribe correct outer `<Product_Type>` tag to all 7 types

- Types: `Collection` , `Bundle`
- `<Product_Type>` fix not complete as of yet

![Screen Shot 2021-07-14 at 1 41 24 PM](https://user-images.githubusercontent.com/35245966/125690082-a93c68c7-106b-427b-a690-987435559faa.png)

Product Type `Collection`, `Bundle` product manually tested using cases from: (https://pds.nasa.gov/data/pds4)

**Comment by jordanpadams** [comment]
@voxparcxls is there anything I can do to get this moving forward? Is there somewhere you are blocked?

**Comment by voxparcxls** [comment]
@jordanpadams For the product_type XML bug, there's nothing large thats blocking me. I just need to retrieve the product selection POST data without it interrupting the following Label pages.

I can use the rest of today to narrow that down. And open PR when it's ready.

**Comment by voxparcxls** [comment]
@jordanpadams 

P.S., the dev version is up on: https://miplvm-docker1.jpl.nasa.gov/plaid  if you would want to try it.

**Comment by voxparcxls** [comment]
## PLAID dev - Docker Deployment

Procedure shows step-by-step directions to deploy PLAID to https://miplvm-docker1.jpl.nasa.gov/plaid 

####  Build and Push PLAID Image to: atifactory.jpl.nasa.gov
1. Build Docker Image
- Move into root of the `/PLAID` directory that holds 'Dockerfile', run command:
Commands:
```
$ docker build --no-cache -t artifactory.jpl.nasa.gov:16001/gov/nasa/jpl/ids/plaid/plaidimage:V1 .
```
```
$ docker run -p 81:8080 plaidimage
```

####  Start PLAID Docker Container

Stay in the same directory and run command:

```
$ docker run -p 81:8080 plaidimage
```

####  Run Docker Compose to Launch PLAID application

Stay in the same directory and run command:

```
$ docker-compose up
```

**Comment by voxparcxls** [comment]
@jordanpadams 

Product_Type names are now prepended to each respective products:
* Product_Observational
* Product_Document
* Product_Context
* Product_Collection
* Product_Bundle

**Test for Product_Context** : https://pds.nasa.gov/data/pds4/context-pds4/node/node.geoscience_1.0.xml

![Screen Shot 2021-07-21 at 2 06 14 PM](https://user-images.githubusercontent.com/35245966/126560149-06a895a4-1afc-43a7-a7f2-d9662ea381d8.png)

**Comment by jordanpadams** [comment]
@voxparcxls looks like there is still something funky going on.

It looks like it is inserting an extra `xml version` and `Product_Context` into the label?

**Comment by voxparcxls** [comment]
Here's the Label product type selection. 


![Screen Shot 2021-07-22 at 12 55 19 AM](https://user-images.githubusercontent.com/35245966/126606368-0a12ce18-582b-425f-9b00-e6ddefe01d23.png)


I'll open a branch for the `Product_Context` fixes.

**Comment by voxparcxls** [comment]
@jordanpadams 

Context Label Output is fixed. Removed resting label error.

- Removed empty `<Discipline_Area></Discipline_Area>` tag from _Product_Context_ labels
- Product_Context no longer has 

![Screen Shot 2021-07-23 at 12 50 13 PM](https://user-images.githubusercontent.com/35245966/126834089-49667be0-e6f5-49bc-a342-86efb8560cf1.png)


Both fixed included in Branch: `PLAID_Issue25`

**Comment by jordanpadams** [comment]
@voxparcxls is this branch ready for a PR?

**Comment by voxparcxls** [comment]
@jordanpadams  

Yes #25 PR is almost ready. I'm writing up the summary for it, and should be opened in a few hours.

Dev-Deployment UPDATE: 
* The new image is pushed to Artifactory: 
`cae-artifactory.jpl.nasa.gov:16001/gov/nasa/jpl/ammos/ids/plaid/plaidimage:V3` 
* Now configuring the DB container for dev, mariadb:10.3.23. (same version as live PLAID)

I will give status on dev deploy this afternoon.

**Comment by voxparcxls** [comment]
@jordanpadams 

Test cases took longer than expected and sidetracked opening the PR. 

Working on publishing the dev deployment ASAP.
